httpie/output/utils.py

```diff
@@ -0,0 +1,39 @@
+import json
+from typing import Generator, Tuple
+
+
+def load_json(data: str) -> Tuple[str, json.JSONDecoder]:
+    """Simple JSON loading from bodies found in `data`.
+    See :func:`find_json_actual_body_and_prefix()` for how bodies are detected.
+
+    """
+    for prefix, body in find_json_actual_body_and_prefix(data):
+        try:
+            return prefix, json.loads(body)
+        except ValueError:
+            pass
+    raise ValueError('Invalid JSON')
+
+
+def find_json_actual_body_and_prefix(data: str) -> Generator[Tuple[str, str], None, None]:
+    """Find potential JSON bodies derived from `data`.
+
+    Sometimes the JSON body is prefixed with a XSSI magic string, specific to the server.
+    In order to be able to display well-formatted JSON, that method will try several
+    approaches to find the actual body of the JSON.
+
+    Yield tuples (data prefix, actual JSON body).
+
+    """
+    # First, the full data.
+    yield '', data
```

Comment:
This should not be done here.  `load_prefixed_json()` should try to parse the whole string on its own before calling `parse_prefixed_json()`.